### ISSUE TITLE
Copy downloaded data instead of re-encoding and allow changing of output format.

### ISSUE THREAD

**Issue by molikuner** [snobu/destreamer#113]
Currently whenever you download a video it gets re-encoded. Most of the times, that's not needed and unnecessary slows down the download on old hardware.
Also it would be nice to be able to specify the output format. Currently it's limited to `mp4`, but there might be people preferring some other format like `mkv`. `mkv` would also have the advantage of allowing almost all codecs being used with it, so the `copy` option for `ffmpeg` would be less likely to fail.

After looking into the code, it seems like it isn't that much effort needed to implement there two things (see my fork (not done yet, but works already)), but as you're already working on doing the download with `aria2c` instead of `ffmpeg` this might not be something you want right now. But since I was able to achieve some good improvements in download speed, so I wanted to let you know.

To sum up:
My proposed changes would be to change the default output format to `mkv` instead of `mp4` and set the default codec to `copy` when launching `ffmpeg`. As the word "default" already implies, I would let the user overwrite there options with some extra arguments. Is that something you want to have right now?

**Comment by snobu** [comment]
If you want to send a PR we are more than happy to do a code review and merge it in.
